geoutils/georaster/raster.py

```diff
@@ -87,6 +86,47 @@ def _default_ndv(dtype: str | np.dtype | type) -> int:
         raise NotImplementedError(f"No default nodata value set for dtype {dtype}")
 
 
+def _load_rio(
+    dataset: rio.io.DatasetReader,
+    bands: int | list[int] | None = None,
+    masked: bool = False,
+    transform: Affine | None = None,
+    shape: tuple[int, int] | None = None,
+    **kwargs: Any,
+) -> np.ndarray | np.ma.masked_array:
+    r"""
+    Load specific bands of the dataset, using rasterio.read().
+    Ensure that self.data.ndim = 3 for ease of use (needed e.g. in show)
+
+    :param bands: The band(s) to load. Note that rasterio begins counting at 1, not 0.
+
+
+    \*\*kwargs: any additional arguments to rasterio.io.DatasetReader.read.
+    Useful ones are:
+    .. hlist::
+    * out_shape : to load a subsampled version
+    * window : to load a cropped version
+    * resampling : to set the resampling algorithm
+    """
+    if transform is not None and shape is not None:
```

Comment:
I don't really understand how shape and transform are to be used. Can you elaborate a bit?

Reply:
Elaborated in b059bfb